org.eclipse.lsp4j/build.gradle

```diff
@@ -17,17 +17,15 @@ sourceCompatibility = '1.8'
 targetCompatibility = '1.8'
 
 dependencies {
-	compile project(":org.eclipse.lsp4j.generator")
-	compile project(":org.eclipse.lsp4j.jsonrpc")
-	testCompile "junit:junit:$versions.junit"
+	api project(":org.eclipse.lsp4j.generator")
```

Comment:
Are you sure you want to expose `generator` dependency to core lsp4j users?

Reply:
did not want to change anything in first step.
https://search.maven.org/artifact/org.eclipse.lsp4j/org.eclipse.lsp4j/0.12.0/jar
dont know how and if it would work without

Reply:
ah. generator is exposed already, sorry.
my point is about visibility of dependencies. If you migrate from `java` plugin to `java-library` you get new `api` configuration. in `java-library` consumer of your library can't see transitive dependencies from `implementation` configuration, but `api` dependencies are visible.
So *mayby* you don't want to just change `compile` dependencies to `api`, but split them: some of them to `api` configuration and some to `implementation` configuration.

Reply:
yes but would prefer to do this in a later step (and by a person with better gradle knowledge) i also dont know if there is people already abusing it.